metricbeat/module/windows/perfmon/config.go

```diff
@@ -0,0 +1,134 @@
+	IgnoreNECounters   bool      `config:"perfmon.ignore_non_existent_counters"`
+	GroupMeasurements  bool      `config:"perfmon.group_measurements_by_instance"`
+	Counters           []Counter `config:"perfmon.counters"`
+	Queries            []Query   `config:"perfmon.queries"`
+	GroupAllCountersTo string    `config:"perfmon.group_all_counter"`
+}
+
+// Counter for the perfmon counters (old implementation deprecated).
+type Counter struct {
+	InstanceLabel    string `config:"instance_label"`
+	InstanceName     string `config:"instance_name"`
+	MeasurementLabel string `config:"measurement_label" validate:"required"`
+	Query            string `config:"query"             validate:"required"`
+	Format           string `config:"format"`
+}
+
+// QueryConfig for perfmon queries. This will be used as the new configuration format
+type Query struct {
+	Name      string         `config:"object" validate:"required"`
+	Field     string         `config:"field"`
+	Instance  []string       `config:"instance"`
+	Counters  []QueryCounter `config:"counters" validate:"required"`
+	Namespace string         `config:"namespace"`
+}
+
+// QueryConfigCounter for perfmon queries. This will be used as the new configuration format
+type QueryCounter struct {
+	Name   string `config:"name" validate:"required"`
+	Field  string `config:"field"`
+	Format string `config:"format"`
+}
+
+func (query *Query) InitDefaults() {
+	query.Namespace = "metrics"
+}
+
+func (counter *QueryCounter) InitDefaults() {
+	form := strings.ToLower(counter.Format)
+	switch form {
+	case "", "float":
+		counter.Format = "float"
+	case "long", "large":
+	}
+}
+
+func (counter *Counter) InitDefaults() {
+	form := strings.ToLower(counter.Format)
+	switch form {
+	case "", "float":
+		counter.Format = "float"
+	case "long", "large":
+	}
+}
+
+func (conf *Config) Validate() error {
+	if len(conf.Counters) == 0 && len(conf.Queries) == 0 {
+		return errors.New("no perfmon counters or queries have been configured")
+	}
+	if len(conf.Counters) > 0 {
+		cfgwarn.Deprecate("8.0", "perfmon.counters configuration option is deprecated and will be removed in the future major version, "+
+			"we advise using the perfmon.queries configuration option instead.")
+	}
+	if len(conf.Queries) > 0 {
+		for _, query := range conf.Queries {
+			if len(query.Counters) == 0 {
+				return errors.Errorf("no perfmon counters have been configured for object %s", query.Name)
+			}
+		}
+	}
+
+	for _, value := range conf.Counters {
+		if !isValidFormat(value.Format) {
+			return errors.Errorf("initialization failed: format '%s' "+
+				"for counter '%s' is invalid (must be float, large or long)",
+				value.Format, value.InstanceLabel)
+		}
+
+	}
+	for _, value := range conf.Queries {
+		for _, q := range value.Counters {
+			if !isValidFormat(q.Format) {
+				return errors.Errorf("initialization failed: format '%s' "+
+					"for counter '%s' is invalid (must be float, large or long)",
+					q.Format, q.Field)
+			}
```

Comment:
Same for this check, it could be done in the `Validate()` method of the Counter type.